src/Service/Startup.cs

```diff
@@ -213,50 +222,61 @@ public void ConfigureServices(IServiceCollection services)
         /// when determining whether to allow introspection requests to proceed.
         /// </summary>
         /// <param name="services">Service Collection</param>
-        private void AddGraphQLService(IServiceCollection services)
+        private void AddGraphQLService(IServiceCollection services, int? depthLimit = null)
         {
-            services.AddGraphQLServer()
-                    .AddHttpRequestInterceptor<DefaultHttpRequestInterceptor>()
-                    .ConfigureSchema((serviceProvider, schemaBuilder) =>
-                    {
-                        GraphQLSchemaCreator graphQLService = serviceProvider.GetRequiredService<GraphQLSchemaCreator>();
-                        graphQLService.InitializeSchemaAndResolvers(schemaBuilder);
-                    })
-                    .AddHttpRequestInterceptor<IntrospectionInterceptor>()
-                    .AddAuthorization()
-                    .AllowIntrospection(false)
-                    .AddAuthorizationHandler<GraphQLAuthorizationHandler>()
-                    .AddErrorFilter(error =>
+            IRequestExecutorBuilder server = services.AddGraphQLServer()
+                .AddHttpRequestInterceptor<DefaultHttpRequestInterceptor>()
+                .ConfigureSchema((serviceProvider, schemaBuilder) =>
+                {
+                    GraphQLSchemaCreator graphQLService = serviceProvider.GetRequiredService<GraphQLSchemaCreator>();
+                    graphQLService.InitializeSchemaAndResolvers(schemaBuilder);
+                })
+                .AddHttpRequestInterceptor<IntrospectionInterceptor>()
+                .AddAuthorization()
+                .AllowIntrospection(false)
+                .AddAuthorizationHandler<GraphQLAuthorizationHandler>();
+
+            // Conditionally adds a maximum depth rule to the GraphQL queries/mutation.
+            // This rule is only added if a positive depth limit is specified, ensuring that the server
+            // enforces a limit on the depth of incoming GraphQL queries/mutation to prevent extremly deep queries
```

Comment:
```suggestion
            // enforces a limit on the depth of incoming GraphQL queries/mutation to prevent extremely deep queries
```

Reply:
This wasn't fixed.

Reply:
i can see it fixed, probably you don't see them because of codespace errors.

Reply:
yep , thanks! codespaces 😔